### ISSUE TITLE
Display Capture Missaligned?

### ISSUE THREAD

**Issue by chrisspiegl** [steveseguin/electroncapture#38]
Hello Everybody, I am just getting to know OBS.Ninja and the Electron Capture helper.
I first was disappointed by the Window Capture delay — thought OBS.Ninja + Electron Capture will not work for me.
Then I read in the Wiki that the Window Capture has a massive delay but Display Capture + Window Crop does not…
Now, however, I tried that and the crop seems to be off by a lot at the top — a screenshot of capture in obs is attached.

![Screen Shot 2020-11-21 at 23 22 19](https://user-images.githubusercontent.com/1096384/99888889-752fba00-2c50-11eb-841d-b24c0cc09336.png)

Basically, there seem to be some strange things going on and I'd love to know if I am missing something? If I have to adjust the crop manually? Or if there is another solution you can provide? I'd love to hear from you.

**Comment by steveseguin** [comment]
Thank  you for the bug report.  That isn't expected.  What resolution is your camera? 16:9 or some other odd aspect ratio?

You can manually resize for now. You can also specify a resolution via the command line, so it will always start at a specify resolution.  

I'll look into this and confirm the issue isn't something new.  

With any luck the Electron Capture app won't be needed much longer, as the OBS guys seem to be working on a solution to bring Browser Source capture back.

-steve

**Comment by chrisspiegl** [comment]
With OBS bringing this into OBS — that would be wonderful.

Thanks for your message, actually the fun part is that the window has the correct aspect ratio but the capture then somehow is missaligned. Not sure what is going on there.

I'll try this out to manage it myself but thanks for your update here.

Is there an issue / forum post I can follow in regards to the Browser Source in the OBS Studio world? I'd love to stay up to date on that.

**Comment by chrisspiegl** [comment]
Additional update, I was able to realign the Display Capture with a manual crop of:

- Crop Top: 95
- Crop Bottom: -95

That helped and fixed it for me. I am happy that the negative bottom crop actually works.

**Comment by steveseguin** [comment]
this is my tracker: https://github.com/steveseguin/obsninja/issues/6

this is their tracker: https://github.com/obsproject/obs-browser/issues/209

The most recent update though was during a conversation I had with this about 2 weeks ago:

![image](https://user-images.githubusercontent.com/2575698/99908965-a8576500-2cb3-11eb-8205-004e426d7342.png)

**Comment by chrisspiegl** [comment]
Thanks, I subscribed to both issues now and I'm looking forward to hear how it goes.

Additionally, I experienced some other things in terms of misalignment with the OBSN.app — The offset I entered manually into the display capture actually changed on me for whatever reason. So my guess is that this may be something happening because I am also using an external display with non-retina and an internal with retina.

Not sure if this is something that is even related to the Electron Capture or would rather be something that needs fixing on the OBS side.

**Comment by steveseguin** [comment]
One thing I see as a possible cause is currently I use "getPrimaryDisplay" to reference the screen DPI and Screen Resolution.  If the app isn't on the primary display, then it will possibly be configured incorrectly. 

I'll need to add logic and event-triggers to detect what display the app is running on and if it changes displays.

**Comment by chrisspiegl** [comment]
That's very possible — on my primary ultra-wide display (non retina) it also works perfectly with the resizing. So that's good.
